modules/hostap/src/supp_events.h

```diff
@@ -41,7 +42,99 @@ enum net_event_wpa_supp_cmd {
 #define NET_EVENT_WPA_SUPP_IFACE_REMOVING					\
 	(_NET_MGMT_WPA_SUPP_EVENT | NET_EVENT_WPA_SUPP_CMD_IFACE_REMOVING)
 
-int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, int status);
+int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, void *status,
+			size_t len);
 int generate_supp_state_event(const char *ifname, enum net_event_wpa_supp_cmd event, int status);
 
+#define REASON_CODE_LEN 32
+#define NM_WIFI_EVENT_STR_LEN 64
+#define ETH_ALEN 6
+
+union supp_mon_event_data {
+	struct supp_event_auth_reject {
+		int auth_type;
+		int auth_transaction;
+		int status_code;
+		uint8_t bssid[ETH_ALEN];
+	} auth_reject;
+
+	struct supp_event_connected {
+		uint8_t bssid[ETH_ALEN];
+		char ssid[WIFI_SSID_MAX_LEN];
+		int id;
+	} connected;
+
+	struct supp_event_disconnected {
+		uint8_t bssid[ETH_ALEN];
+		int reason_code;
+		int locally_generated;
+	} disconnected;
+
+	struct supp_event_assoc_reject {
+		int status_code;
+		int reason_code;
+	} assoc_reject;
+
+	struct supp_event_temp_disabled {
+		int id;
+		char ssid[WIFI_SSID_MAX_LEN];
+		unsigned int auth_failures;
+		unsigned int duration;
+		char reason_code[REASON_CODE_LEN];
+	} temp_disabled;
+
+	struct supp_event_reenabled {
+		int id;
+		char ssid[WIFI_SSID_MAX_LEN];
+	} reenabled;
+
+	struct supp_event_bss_added {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_added;
+
+	struct supp_event_bss_removed {
+		unsigned int id;
+		uint8_t bssid[ETH_ALEN];
+	} bss_removed;
+
+	struct supp_event_network_added {
+		unsigned int id;
+	} network_added;
+
+	struct supp_event_network_removed {
+		unsigned int id;
+	} network_removed;
+
+	char supp_event_str[NM_WIFI_EVENT_STR_LEN];
+};
+
+enum supp_event_num {
+	WPA_SUPP_WIFI_CMD_NM_EVENT = NET_EVENT_WPA_SUPP_CMD_COUNT,
+	WPA_SUPP_CONNECTED,
+	WPA_SUPP_DISCONNECTED,
+	WPA_SUPP_ASSOC_REJECT,
+	WPA_SUPP_AUTH_REJECT,
+	WPA_SUPP_TERMINATING,
+	WPA_SUPP_SSID_TEMP_DISABLED,
+	WPA_SUPP_SSID_REENABLED,
+	WPA_SUPP_SCAN_STARTED,
+	WPA_SUPP_SCAN_RESULTS,
+	WPA_SUPP_SCAN_FAILED,
+	WPA_SUPP_BSS_ADDED,
+	WPA_SUPP_BSS_REMOVED,
+	WPA_SUPP_NETWORK_NOT_FOUND,
+	WPA_SUPP_NETWORK_ADDED,
+	WPA_SUPP_NETWORK_REMOVED
+};
+
+#define NET_EVENT_WPA_SUPP_NM_EVENT					\
+	(_NET_MGMT_WPA_SUPP_EVENT | WPA_SUPP_WIFI_CMD_NM_EVENT)
+
+struct supp_status_data {
```

Comment:
This isn't `status`, so, `supp_int_event_data` ?

Reply:
Done